src/MouseAdapter.js

```diff
@@ -309,20 +383,196 @@ define(function(require) {
         _onMouseWheel: function(event) {
             // If we haven't started a mouse wheel for a bit, publish a start event.
             if (!this._wheeling) {
-                this.onMouseWheelStart.dispatch([{
-                    distance: { x: 0, y: 0 },
-                    source: event
-                }]);
-                this._wheeling = true;
+                this._dispatchMouseWheelStart(event);
             }
 
             // Normalize the mouse wheel event and publish.
             var normalizedEvent = normalizeEvent(event);
+            this._detectUserScroll(normalizedEvent);
             this.onMouseWheel.dispatch([normalizedEvent]);
 
             // Debounce mouse wheels so that an end event is published 50ms
             // after the last mousewheel occured.
+            this._debouncedDispatchMouseWheelEnd(event);
+        },
+
+        /**
+         * Dispatches the MouseWheel start event, and handles related controls.
+         * @param {object} event
+         * @private
+         */
+        _dispatchMouseWheelStart: function(event) {
+            this.onMouseWheelStart.dispatch([{
+                distance: { x: 0, y: 0 },
+                source: event
+            }]);
+            this._wheeling = true;
+        },
+
+        /**
+         * Dispatches the MouseWheel end event, and handles related controls.
+         * @param {object} event
+         * @private
+         */
+        _dispatchMouseWheelEnd: function(event) {
+
+            // Reset the delta array on MouseEnd to prevent false positives between
+            // natural end and start events.
+            deltaArray = [[],[]];
+
+            this.onMouseWheelEnd.dispatch([{
+                distance: { x: 0, y: 0 },
+                source: event
+            }]);
+            this._wheeling = false;
+        },
```

Comment:
I'm guessing that the distances for these events are `0` because all events follow a uniform interface and those values just aren't relevant for these two events, right?